### PR TITLE
Update appveyor image to Visual Studio 2017

### PR DESCRIPTION
**Pull request by Reflexe**

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -16,8 +16,9 @@ build_script:
   - cd build
   - ps: $env:CMAKE_PLATFORM="$(if ($env:PLATFORM -eq 'x64') { 'x64' } else { '' })"
   - ps: $env:QT_SUFFIX="$(if ($env:PLATFORM -eq 'x64') { '_64' } else { '' })"
-  - cmake -DUSE_COMPILE_CACHE=ON -DCACHE_TOOL=%APPVEYOR_BUILD_FOLDER%/clcache.4.1.0/clcache-4.1.0/clcache.exe -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_PREFIX_PATH=c:/Qt/5.9/msvc2015%QT_SUFFIX%;c:/tools/vcpkg/installed/%PLATFORM%-windows -DCMAKE_GENERATOR_PLATFORM="%CMAKE_PLATFORM%" ..
+  - cmake -DUSE_COMPILE_CACHE=ON -DCACHE_TOOL=%APPVEYOR_BUILD_FOLDER%/clcache.4.1.0/clcache-4.1.0/clcache.exe -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_PREFIX_PATH=c:/Qt/5.12/msvc2017%QT_SUFFIX%;c:/tools/vcpkg/installed/%PLATFORM%-windows -DCMAKE_GENERATOR_PLATFORM="%CMAKE_PLATFORM%" ..
   - cmake --build . -- /maxcpucount:4
   - cmake --build . --target tests
 cache:
   - c:/tools/vcpkg/installed
+  - "%HOME%/clcache"
```
Comment (PhysSong) [inline .appveyor.yml]: Could you move this to a separate pull request?

**Comment by PhysSong** [comment]
It seems like Vcpkg shipped with the Visual Studio 2015 image is outdated. Can we ask Appveyor updating old broken Vcpkg for that image?

**Comment by Reflexe** [comment]
> It seems like Vcpkg shipped with the Visual Studio 2015 image is outdated. Can we ask Appveyor updating old broken Vcpkg for that image?

I'll try that.

**Comment by Reflexe** [comment]
> It seems like Vcpkg shipped with the Visual Studio 2015 image is outdated. Can we ask Appveyor updating old broken Vcpkg for that image?

Is there a problem with just upgrading to visual studio 2017?

**Comment by PhysSong** [comment]
> Is there a problem with just upgrading to visual studio 2017?

I'm not sure. @lukas-w chose VS2017, so he might be able to answer this.

**Comment by lukas-w** [comment]
> @lukas-w chose ~VS2017~ [VS2015], so he might be able to answer this.

I don't remember a specific reason, possibly compatibility with some dependency. Let's just upgrade.
